### ISSUE TITLE
Clean up use of unittest assert functions in twisted/words

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8394]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8394|
|Type|enhancement|
|Created|2016-06-03 02:18:26Z|
|Branch|https://github.com/twisted/twisted/tree/8394-rodrigc-assert|

This ticket is split out from [ticket:8371] because the patch there was too big to review.

The files in twisted/words
should have their use of unittest assert functions cleaned up so that:

* the most appropriate unittest assert function is used appropriately in each file
* use of deprecated aliases should be eliminated, to reduce deprecation
  warnings and errors when moving to Python3 compatibility

This has some pointers about deprecated unittest assert aliases:

See: https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

|= Method Name =|= Deprecated alias(es) =|
|---------------|------------------------|
| assertEqual() | failUnlessEqual, assertEquals |
| assertNotEqual() | failIfEqual |
| assertTrue() | failUnless, assert_ |
| assertFalse() | failIf |
| assertRaises() |	failUnlessRaises |
| assertAlmostEqual() | failUnlessAlmostEqual  |
| assertNotAlmostEqual() | failIfAlmostEqual  |

<details><summary>Searchable metadata</summary>

```
trac-id__8394 8394
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8394_rodrigc_assert 8394-rodrigc-assert
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464920306983970 1464920306983970
changetime__1466479051141215 1466479051141215
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/110

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for your contribution, rodrigc.

As with [#8376](https://github.com/twisted/twisted/issues/8376), the changes to these assertion functions are over-broad.  These are a lot of changes to look over, but please re-submit more minimal changes that just address the necessary bits to migrate to python 3.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc removed owner|
|-|-|

OK, I uploaded a more minimal new diff, which just removes deprecated aliases.
See:
https://github.com/twisted/twisted/pull/110

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

Changes look good, but insread of `assertTrue(not something)` or `assertTrue(isinstance())` I prefer to have the dedicated assertFalse and assertIsInstance methods.

I have updated the patch to use those methods.

Since I have also changed this branch I am leaving this for review so that another dev can take a look.

branch sent to buildbot

thanks Craig for your contribution.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_5"></a>|@wsanchez set owner to @rodrigc|
|-|-|

Reviewed, good to merge.  (Agree with Adi.)

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset a00e2fa877cfd3b91c295f25d6f95727357a361a
```
#!CommitTicketReference repository="" revision="a00e2fa877cfd3b91c295f25d6f95727357a361a"
Merge '8394-rodrigc-assert': Cleanup use of unittest assertion functions

Author: rodrigc, adiroiban
Reviewer: wsanchez, adiroiban
Fixes: #8394
```
